src/components/album/AlbumsList.js

```diff
@@ -22,41 +24,51 @@ import {
 
 const AlbumsList = () => {
   const [albums, setAlbums] = useState([]); // for albums
-  const [limit, setLimit] = useState(1); // for pagination
+  const limit = 12;
+  // const [limit, setLimit] = useState(12); // for pagination
   const [searchType, setSearchType] = useState('albumName'); //default value for select input
   const [searchTerm, setSearchTerm] = useState(''); // for search input filed value
   const [message, setMessage] = useState(''); // for not found message
   const [errorMessage, setErrorMessage] = useState(''); // for error message
   const [wishListId, setWishListId] = useState();
+  const [currentPage, setCurrentPage] = useState(1); // Track current page
+  // const [lastSearchTerm, setLastSearchTerm] = useState('');
 
   //make an API call with search values to backend and return the result
-  const fetchAlbums = async (searchType, searchTerm, limit) => {
-    try {
-      const response = await axios.get(
-        `${process.env.REACT_APP_API_BASE_PATH}/albums/filter`,
-        {
-          params: {
-            limit,
-            [searchType]: searchTerm,
-          },
+  const fetchAlbums = useCallback(
+    async (searchType, searchTerm) => {
+      try {
+        const offset = (searchTerm === '' ? currentPage - 1 : 0) * limit;
+        // console.log(currentPage - 1 + ' * ' + limit + ' = ' + offset);
+        // console.log('searchType:' + searchType + ' ; searchTerm:' + searchTerm);
+
+        const response = await axios.get(
+          `${process.env.REACT_APP_API_BASE_PATH}/albums/filter`,
+          {
+            params: {
+              limit: limit,
+              [searchType]: searchTerm,
+              offset,
+            },
+          }
+        );
+        const searchResult = response.data.albums;
+        setAlbums(searchResult);
+        if (searchResult.length === 0) {
+          setMessage('No result found');
         }
-      );
-      const searchResult = response.data.albums;
-      setAlbums(searchResult);
-      if (searchResult.length === 0) {
-        setMessage('No result found');
+      } catch (error) {
+        console.error('Error fetching albums:', error);
+        setErrorMessage(error.message);
+        setOpen(true);
       }
-    } catch (error) {
-      console.error('Error fetching albums:', error);
-      setErrorMessage(error.message);
-      setOpen(true);
-    }
-  };
-
-  //display 10 albums when first user visit this page
+    },
+    [currentPage, limit]
+  );
+  //display 12 albums when first user visit this page
   useEffect(() => {
-    fetchAlbums('albumName', '', 10);
-  }, []);
+    fetchAlbums('albumName', '');
+  }, [fetchAlbums, currentPage]); // Refetch albums when the page changes
```

Comment:
```suggestion
  useEffect(() => {
    fetchAlbums(searchType, searchTerm);
  }, [fetchAlbums, currentPage]); // Refetch albums when the page changes
```

This will ensure that we refetch, with the latest values of searchType and searchTerm

Reply:
I did this, but now I got this error:
  Line 80:6:  React Hook useEffect has missing dependencies: 'searchTerm' and 'searchType'.

Reply:
I added them to a dependency list:   
`}, [fetchAlbums, currentPage, searchTerm, searchType]);
`

Reply:
Looks good; that is the correct next step there 👍 .